modules/auxiliary/scanner/http/iis_shortname_scanner.rb

```diff
@@ -0,0 +1,280 @@
+  end
+
+  def scanner
+    while !@queue_ext.empty?
+      f = @queue_ext.pop
+      url = f.split(':')[0]
+      ext = f.split(':')[1]
+      status = get_incomplete_status(url, "*" , "1" , ext)
+      if status == 404 && ext.size == 3
+        @found3.each do |x|
+          if get_complete_status(url, x , ext) == 404
+            @files << "#{url}~#{x}.#{ext}"
+          end
+        end
+      elsif status == 404 and ext.size < 4
+        @found3.each do |x|
+          if get_complete_status(url, x , ext) == 404
+            @files << "#{url}~#{x}.#{ext}"
+          end
+        end
+        for c in @found2
+          @queue_ext << (f + c )
+        end
+      else
+      end
+    end
+  end
+
+  def scan
+    while !@queue.empty?
+      url = @queue.pop
+      status = get_status(url , "1" , "*")
+      if url.size == 6 && status == 404
+        @found3.each do |x|
+          if get_status(url , x , "") == 404
+            @dirs << "#{url}~#{x}"
+          end
+        end
+        for ext in @found2
+          @queue_ext << ( url + ':' + ext )
+          @threads << Thread.new { scanner }
+        end
+      elsif status == 404
+        @found3.each do |x|
+          if get_complete_status(url, x , "") == 404
+            @dirs << "#{url}~#{x}"
+            break
+          end
+        end
+        if get_incomplete_status(url, "" , "1" , "") == 404
+          for ext in @found2
+            @queue_ext << ( url + ':' + ext )
+            @threads << Thread.new { scanner }
+          end
+        elsif url.size   < 6
+          for c in @found
+            @queue  <<(url +c)
+          end
+        end
+      end
+    end
+  end
+
+  def reduce
+    for c in @alpha.chars
+      res = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'],"*#{c}*~1*"),
+        'method' => @verb
+      })
+      if res.code == 404
+        @found << c
+      end
+    end
+  end
+
+  def ext
+    for c in @alpha.chars
+      res = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'],"*~1.*#{c}*"),
+        'method' => @verb
+      })
+      if res.code == 404
+        @found2 << c
+      end
+    end
+  end
+
+  def dup
+    array = [*('1'..'9')]
+    array.each do |c|
+      res = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'],"*~#{c}.*"),
+        'method' => @verb
+      })
+      if res.code == 404
+        @found3 << c
+      end
+    end
+  end
+
+  def run
+    unless is_vul
+      print_status("Target is not vulnerable, or no shortname scannable files are present.")
+      return
+    end
+      print_status("Scanning in progress...")
+      @threads << Thread.new { reduce }
+      @threads << Thread.new { dup }
+      @threads << Thread.new { ext }
+      @threads.each(&:join)
+
+      for c in @found
+        @queue << c
+      end
+
+      20.times {
+        @threads << Thread.new { scan }
+      }
+
+      sleep(1) until @queue_ext.empty?
+
+      @threads.each(&:join)
+      if @dirs.empty?
```

Comment:
The URL scheme should be generated depending on whether SSL is enabled, rather than hardcoded `http://`.

Also, golfed:

```ruby
      proto = datastore['SSL'] ? 'https' : 'http'

      if @dirs.empty?
        print_status("No directories were found")
      else
        print_good("Found #{@dirs.size} directories")
        @dirs.each do |x|
          print_line("#{proto}://#{datastore['RHOST']}#{datastore['PATH']}#{x}")
        end
      end

      if @files.empty?
        print_status("No files were found")
      else
        print_good("Found #{@files.size} files")
        @files.each do |x|
          print_line("#{proto}://#{datastore['RHOST']}#{datastore['PATH']}#{x}")
        end
      end
```